spring-cloud-starter-stream-sink-sftp/src/main/java/org/springframework/cloud/stream/app/sftp/sink/SftpSinkConfiguration.java

```diff
@@ -58,4 +58,25 @@ public IntegrationFlow ftpInboundFlow(SftpSinkProperties properties, SessionFact
 			.get();
 	}
 
+	@Bean
+	@ConditionalOnMissingBean
+	public SessionFactory<LsEntry> sftpSessionFactory(SftpSinkProperties properties) {
+		DefaultSftpSessionFactory sftpSessionFactory = new DefaultSftpSessionFactory();
+		sftpSessionFactory.setHost(properties.getFactory().getHost());
+		sftpSessionFactory.setPort(properties.getFactory().getPort());
+		sftpSessionFactory.setUser(properties.getFactory().getUsername());
+		sftpSessionFactory.setPassword(properties.getFactory().getPassword());
+		sftpSessionFactory.setAllowUnknownKeys(properties.getFactory().isAllowUnknownKeys());
+		if (properties.getFactory().getKnownHostsExpression() != null) {
+			sftpSessionFactory.setKnownHosts("#{" + properties.getFactory().getKnownHostsExpression() + "}");
```

Comment:
Well, in XD that looks like:
```
<beans:property name="knownHosts" value="#{${knownHostsExpression:null}}" />
```
Which is really bean initialization SpEL, not a target property value like in this case.
There it has a description like:
```
@ModuleOption("a SpEL expresssion location of known hosts file; required if 'allowUnknownKeys' is false; "
			+ "examples: systemProperties[\"user.home\"]+\"/.ssh/known_hosts\", \"/foo/bar/known_hosts\"")
	public void setKnownHostsExpression(String knownHostsExpression) {
```
Which really makes sense. We can come up with something similar in Java config via `@Value`